types/src/v0/v0_1/instance_state.rs

```diff
@@ -20,30 +23,63 @@ pub enum UpgradeType {
     ChainConfig { chain_config: ChainConfig },
 }
 
-/// Represents the  upgrade config including the type of upgrade and upgrade parameters for hotshot config.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct Upgrade {
-    /// The view at which the upgrade is proposed.
-    ///
-    /// Note: Voting for the proposal begins before the upgrade is formally proposed.
-    /// In our implementation, `start_proposing_view` is set to `1`` for all upgrades,
-    /// so if an upgrade is planned then the voting starts as soon as node is started.
-    #[serde(rename = "view")]
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct TimeBasedUpgrade {
+    pub start_proposing_time: u64,
+    pub stop_proposing_time: u64,
+    pub start_voting_time: Option<u64>,
+    pub stop_voting_time: Option<u64>,
```

Comment:
Instead of `u64` we should use the `Timestamp` type (from genesis.rs) so that these parse from a nice human-readable datetime

Reply:
changed